packages/components/src/components/Checkbox/Checkbox.stories.tsx

```diff
@@ -0,0 +1,154 @@
+import { Meta, StoryObj } from '@storybook/react'
+import { View } from 'react-native'
+import React, { useState } from 'react'
+
+import { Checkbox, CheckboxProps } from './Checkbox'
+import { generateDocs } from '../../utils/storybook'
+
+const meta: Meta<CheckboxProps> = {
+  title: 'Checkbox',
+  component: Checkbox,
+  decorators: [
+    (Story) => (
+      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+        {Story()}
+      </View>
+    ),
+  ],
+  parameters: {
+    design: [
+      {
+        name: 'Figma component overview',
+        type: 'figma',
+        url: 'REPLACE',
+      },
+      {
+        name: 'Figma examples',
+        type: 'figma',
+        url: 'REPLACE',
+      },
+    ],
+    docs: generateDocs({
+      name: 'Checkbox',
+      docUrl: 'REPLACE',
+    }),
+  },
+}
+
+export default meta
+
+type Story = StoryObj<CheckboxProps>
+
+const statefulComponentRenderer = (props: CheckboxProps) => {
+  const {
+    description,
+    descriptionA11y,
+    error,
+    errorA11y,
+    header,
+    headerA11y,
+    hint,
+    hintA11y,
+    indeterminate,
+    label,
+    labelA11y,
+    required,
+    tile,
+  } = props
+
+  const [checked, setChecked] = useState(false)
```

Comment:
Wonder if doing the same with indeterminate will fix the weirdness observed in the one recording you had turning it on.